pkg/apis/core/validation/shoot.go

```diff
@@ -376,6 +376,16 @@ func validateKubeProxyModeUpdate(newConfig, oldConfig *core.KubeProxyConfig, ver
 	if ok, _ := versionutils.CheckVersionMeetsConstraint(version, "< 1.16"); ok {
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newMode, oldMode, fldPath.Child("mode"))...)
 	}
+	// The enabled flag is immutable for now to ensure that the networking extensions have time to adapt to it.
+	newEnabled := true
+	oldEnabled := true
+	if newConfig != nil {
+		newEnabled = *newConfig.Enabled
+	}
+	if oldConfig != nil {
+		oldEnabled = *oldConfig.Enabled
+	}
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newEnabled, oldEnabled, fldPath.Child("enabled"))...)
```

Comment:
Can you add a corresponding test case to https://github.com/gardener/gardener/blob/30dc4e926037b7bb3e5a7f9d8d130fdfccc8ee60/pkg/apis/core/validation/shoot_test.go#L1748?

Reply:
Test introduced (and bugs in validation fixed).